source/fab/builder.py

```diff
@@ -129,10 +141,20 @@ def __init__(self,
             'gfortran',
             ['-c', '-J', str(workspace)] + fc_flags.split(), workspace
         )
-        # TODO: Linker class is probably generic not Fortran specific
-        # now that we are passing in the command to use?
-        fortran_linker = FortranLinker(
-            'gfortran', ld_flags.split(), workspace, exec_name
+
+        header_analyser = HeaderAnalyser(workspace)
+        c_pragma_injector = CPragmaInjector(workspace)
+        c_preprocessor = CPreProcessor(
+            'cpp', ['-P'], workspace
```

Comment:
I think we picked up the `-P` from me copying the UM configs and then I blindly copied it from the Fortran PreProcessor class into the C one.  It probably existed for Fortran because the compiler doesn't like the linemarkers, so I think you're right that in the C case it should be okay with them.  I'll take it out and see if it still works